crates/common_enums/src/enums.rs

```diff
@@ -757,6 +757,7 @@ pub enum EventType {
     DisputeChallenged,
     DisputeWon,
     DisputeLost,
+    UnsupportedEvent,
```

Comment:
Instead of doing this, can we throw some other error like `UnprocessableEntity` - 422 instead of `InternalServerError`